assets/pages/regular-contributions/components/contributionsGuestCheckoutContainer.jsx

```diff
@@ -32,29 +30,25 @@ const mapDispatchToProps = (dispatch: Dispatch<CheckoutAction>) => ({
   onBackClick: () => {
     dispatch(setStage('checkout'));
   },
+
   onNextButtonClick: (
     contributionType: Contrib,
     isSignedIn: boolean,
     userTypeFromIdentityResponse: UserTypeFromIdentityResponse,
+    paymentMethod: PaymentMethod,
   ) => {
-    const canContribute =
-      canContributeWithoutSigningIn(contributionType, isSignedIn, userTypeFromIdentityResponse)
-      || isSignedIn;
-
-    const componentId = `opf-${contributionType}-submit`;
     const formName = 'regular-contrib__checkout-form';
-    if (formIsValid(formName)) {
-      if (canContribute) {
-        dispatch(setStage('payment'));
-        trackCheckoutSubmitAttempt(componentId, `opf-allowed-for-user-type-${userTypeFromIdentityResponse}`);
-      } else {
-        trackCheckoutSubmitAttempt(componentId, `opf-blocked-because-user-type-is-${userTypeFromIdentityResponse}`);
-      }
-    } else {
-      trackCheckoutSubmitAttempt(componentId, `opf-blocked-because-form-not-valid${invalidReason(getForm(formName))}`);
-    }
-
-    dispatch(setCheckoutFormHasBeenSubmitted());
+    onFormSubmit(
+      'opf',
+      paymentMethod,
+      contributionType,
+      getForm(formName),
+      isSignedIn,
+      userTypeFromIdentityResponse,
+      () => {}, // we don't need this for the old flow
+      () => { dispatch(setCheckoutFormHasBeenSubmitted()); },
+      () => { dispatch(setStage('payment')); },
```

Comment:
here's an example where having names would add value - for these last three functions

for those where the names are the same as the parameter names (`paymentMethod`, `contributionType`), you wouldn't have any repetition as you can just use the shorthand object syntax.